### PR TITLE
openarena: add module and test

### PR DESCRIPTION
**Pull request by tomfitzhenry**
###### Motivation for this change

A new NIxOS module and test for running a dedicated OpenArena server.

Motivation is 🔫🚀💥, I suppose.

###### Things done

nix-build on the test passes. To observe the test failing, remove the "+map oa_dm1" flag, which will cause `openarena.service` to not respond to `getchallenge`.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

n/a

### CODE REVIEW

File: nixos/modules/services/games/openarena.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  name = "OpenArena";
```
Comment (dasJ) [inline nixos/modules/services/games/openarena.nix]: Why this abstraction?

File: nixos/modules/services/games/openarena.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      enable = mkEnableOption name;
+
+      openPorts = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to open firewall ports for " + name;
+      };
+
+      extraFlags = mkOption {
+        type = types.string;
```
Comment (dasJ) [inline nixos/modules/services/games/openarena.nix]: `types.string` is deprecated. Please use `types.listOf types.str` and `concatStringsSep " " cfg.extraFlags` below

**Review by dasJ — CHANGES_REQUESTED** [review]

File: nixos/modules/services/games/openarena.nix (hunk 1 above)
Comment (tomfitzhenry) [reply nixos/modules/services/games/openarena.nix]: Over-eager variable extraction? I've inlined this variable now.

**Review by dasJ — APPROVED** [review]

**Comment by aanderse** [comment]
@GrahamcOfBorg test openarena

**Comment by tomfitzhenry** [comment]
@aanderse Done: I have added myself to `maintainers-list.nix` and rebased that to be the first commit of this PR.

File: nixos/modules/services/games/openarena.nix [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          # Load a map. Mandatory for clients to be able to connect.
+          "+map oa_dm1"
+        ];
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = with pkgs; [
+      openarena
```
Comment (aanderse) [inline nixos/modules/services/games/openarena.nix]: Why is this required? Generally we don't add the package to the system environment unless explicitly required.

**Review by aanderse — CHANGES_REQUESTED** [review]

File: nixos/modules/services/games/openarena.nix (hunk 3 above)
Comment (tomfitzhenry) [reply nixos/modules/services/games/openarena.nix]: Done. TIL that isn't required.

**Review by aanderse — APPROVED** [review]
LGTM :+1: 

Thanks for your contribution :tada:

**Comment by aanderse** [comment]
@GrahamcOfBorg eval
